File: src/main/java/io/jsql/sql/handler/replication_statement/RESET_MASTER.java
package io.jsql.sql.handler.replication_statement;

import io.jsql.config.ErrorCode;
import io.jsql.sql.OConnection;

/**
 * Created by 长宏 on 2017/3/27.
 */
public class RESET_MASTER {
    public static boolean isMe(String sql) {
        String[] strings = sql.split("\\s+");
        return strings.length > 2 && strings[0].equalsIgnoreCase("reset") && strings[1].equalsIgnoreCase("master");
    }

    public static void handle(String sql, OConnection c) {
        c.writeErrMessage(ErrorCode.ER_NOT_SUPPORTED_YET, "暂未支持");
    }
}

File: src/main/java/io/jsql/sql/handler/replication_statement/RESET_MASTER.kt
/*
 * Java-based distributed database like Mysql
 */

package io.jsql.sql.handler.replication_statement

import io.jsql.config.ErrorCode
import io.jsql.sql.OConnection

/**
 * Created by 长宏 on 2017/3/27.
 */
object RESET_MASTER {
    fun isMe(sql: String): Boolean {
        val strings = sql.split("\\s+".toRegex()).dropLastWhile { it.isEmpty() }.toTypedArray()
        return strings.size > 2 && strings[0].equals("reset", ignoreCase = true) && strings[1].equals("master", ignoreCase = true)
    }

    fun handle(sql: String, c: OConnection) {
        c.writeErrMessage(ErrorCode.ER_NOT_SUPPORTED_YET, "暂未支持")
    }
}
